arrow/array.go

```diff
@@ -127,3 +127,15 @@ type Array interface {
 	// When the reference count goes to zero, the memory is freed.
 	Release()
 }
+
+// ValueType is a generic constraint for valid Arrow primitive types
+type ValueType interface {
+	bool | FixedWidthType | string | []byte
```

Comment:
FixedWidthType sticks out here but I guess it's because there's no equivalent Golang type; however does it make sense to also allow StringType etc. here?

Reply:
`FixedWidthType` is not `FixedWidthDataType`, `FixedWidthType` is a generic constraint while `FixedWidthDataType` is the actual interface. See `type_traits.go`

in this case, `FixedWidthType` is just short-hand for all of the signed and unsigned int types, float types, decimal types, and temporal types.

Notice we have `| string | []byte` which would include all of the string and binary types too.

Reply:
ah I see